platform/settings.cpp

```diff
@@ -17,375 +17,407 @@
+    LOG(LWARNING, ("Loading settings:", ex.Msg()));
+  }
+}
 
-  void StringStorage::Save() const
+void StringStorage::Save() const
+{
+  try
   {
-    try
+    FileWriter file(GetPlatform().SettingsPathForFile(SETTINGS_FILE_NAME));
+    for (auto const & value : m_values)
     {
-      FileWriter file(GetPlatform().SettingsPathForFile(SETTINGS_FILE_NAME));
-      for (auto const & value : m_values)
-      {
-        string line(value.first);
-        line += DELIM_CHAR;
-        line += value.second;
-        line += "\n";
-        file.Write(line.data(), line.size());
-      }
-    }
-    catch (RootException const & ex)
-    {
-      // Ignore all settings saving exceptions.
-      LOG(LWARNING, ("Saving settings:", ex.Msg()));
+      string line(value.first);
+      line += DELIM_CHAR;
+      line += value.second;
+      line += "\n";
+      file.Write(line.data(), line.size());
     }
   }
-
-  StringStorage & StringStorage::Instance()
+  catch (RootException const & ex)
   {
-    static StringStorage inst;
-    return inst;
+    // Ignore all settings saving exceptions.
+    LOG(LWARNING, ("Saving settings:", ex.Msg()));
   }
+}
 
-  void StringStorage::Clear()
-  {
-    lock_guard<mutex> guard(m_mutex);
-    m_values.clear();
-    Save();
-  }
+StringStorage & StringStorage::Instance()
+{
+  static StringStorage inst;
+  return inst;
+}
 
-  bool StringStorage::GetValue(string const & key, string & outValue) const
-  {
-    lock_guard<mutex> guard(m_mutex);
+void StringStorage::Clear()
+{
+  lock_guard<mutex> guard(m_mutex);
+  m_values.clear();
+  Save();
+}
 
-    auto const found = m_values.find(key);
-    if (found == m_values.end())
-      return false;
+bool StringStorage::GetValue(string const & key, string & outValue) const
+{
+  lock_guard<mutex> guard(m_mutex);
 
-    outValue = found->second;
-    return true;
-  }
+  auto const found = m_values.find(key);
+  if (found == m_values.end())
+    return false;
 
-  void StringStorage::SetValue(string const & key, string && value)
-  {
-    lock_guard<mutex> guard(m_mutex);
+  outValue = found->second;
+  return true;
+}
 
-    m_values[key] = move(value);
-    Save();
-  }
+void StringStorage::SetValue(string const & key, string && value)
+{
+  lock_guard<mutex> guard(m_mutex);
 
-  void StringStorage::DeleteKeyAndValue(string const & key)
-  {
-    lock_guard<mutex> guard(m_mutex);
+  m_values[key] = move(value);
+  Save();
+}
 
-    auto const found = m_values.find(key);
-    if (found != m_values.end())
-    {
-      m_values.erase(found);
-      Save();
-    }
+void StringStorage::DeleteKeyAndValue(string const & key)
+{
+  lock_guard<mutex> guard(m_mutex);
+
+  auto const found = m_values.find(key);
+  if (found != m_values.end())
+  {
+    m_values.erase(found);
+    Save();
   }
+}
 
 ////////////////////////////////////////////////////////////////////////////////////////////
 
-  template <> string ToString<string>(string const & str)
-  {
-    return str;
-  }
+template <>
```

Comment:
I don’t think it’s possible to put template specializations into an anonymous namespace.